language/extensions/move-table-extension/src/lib.rs

```diff
@@ -218,157 +220,129 @@ impl TableData {
         key_ty: &Type,
         value_ty: &Type,
     ) -> PartialVMResult<&mut Table> {
-        if let Entry::Vacant(e) = self.tables.entry(handle) {
-            let key_layout = get_type_layout(context, key_ty)?;
-            let value_layout = get_type_layout(context, value_ty)?;
-            let table = Table {
-                handle,
-                key_layout,
-                value_layout,
-                content: Default::default(),
-            };
-            e.insert(table);
-        }
-        Ok(self.tables.get_mut(&handle).unwrap())
+        Ok(match self.tables.entry(handle) {
+            Entry::Vacant(e) => {
+                let key_layout = get_type_layout(context, key_ty)?;
+                let value_layout = get_type_layout(context, value_ty)?;
+                let table = Table {
+                    handle,
+                    key_layout,
+                    value_layout,
+                    content: Default::default(),
+                };
+                e.insert(table)
+            }
+            Entry::Occupied(e) => e.into_mut(),
+        })
     }
 }
 
 impl Table {
-    /// Inserts a value into a table.
```

Comment:
@wrwg I had to delete all these helpers since they interacted poorly with the gas metering code. Found it much easier to just move the logic into the native functions instead of passing the gas parameters around.